Components/Queues/AsyncQueueConsumer.php

```diff
@@ -4,90 +4,164 @@
 
 namespace Smartbox\Integration\FrameworkBundle\Components\Queues;
 
-use Psr\Log\LoggerAwareInterface;
-use Psr\Log\LoggerAwareTrait;
-use Smartbox\Integration\FrameworkBundle\Components\Queues\Handler\AmqpQueueHandler;
-use Smartbox\Integration\FrameworkBundle\Core\Consumers\ConsumerInterface;
-use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use PhpAmqpLib\Message\AMQPMessage;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\AsyncQueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\QueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\AbstractAsyncConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
 use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
 use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
-use Smartbox\Integration\FrameworkBundle\Service;
 
-class AsyncQueueConsumer extends Service implements ConsumerInterface, LoggerAwareInterface
+/**
+ * Class AsyncQueueConsumer
+ */
+class AsyncQueueConsumer extends AbstractAsyncConsumer
 {
-    use IsStopableConsumer;
     use UsesSmartesbHelper;
     use UsesSerializer;
-    use LoggerAwareTrait;
     use UsesExceptionHandlerTrait;
 
     /**
-     * @var QueueManager
+     * Consumer identifier name.
      */
-    private $manager;
+    const CONSUMER_TAG = 'amqp-consumer-%s-%s';
 
     /**
-     * @var string
+     * {@inheritdoc}
      */
-    private $format;
+    protected function initialize(EndpointInterface $endpoint)
+    {
+        if (!($driver = $this->getQueueDriver($endpoint))->isConnected()) {
+            $driver->connect();
+        }
+    }
```

Comment:
I will bring the logic to know if the driver is connect or not to the driver itself, like you did in the `QueueConsumer.php` and the StompDriver. Doing that we will have both drivers with the same responsibility, and more important, the connection won't be created each time a new message is produced. Nowadays on the `QueueProducer` we are doing 
```
 $queueDriver->connect();
```

Reply:
I agree. I applied the changes regards this and the next comment to let the driver with the responsibility to check if there is any connection or not.